lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -1418,8 +1418,32 @@ SmallVector<Direction> PrimitiveOp::portDirections() {
 
```

Comment:
One thing I'm stuck on - won't this change the outputted IR for port MLIR attributes, i.e., should this have a test?

Your comment in the description seems to say otherwise, so let me know if I'm misunderstanding.

Reply:
No, it won't change the output IR. Instances of components, primitive, etc, always pull their port attributes from the definition of what is being instantiated. For example, looking at [a test component instance](https://github.com/llvm/circt/blob/d638a275e2c8df99f3cfe7012555648085d7888a/test/Dialect/Calyx/emit.mlir#L98) we can see that it has no attached port level attributes, even though it has clk and done ports. These attributes are pulled from the referenced component definition.

What I'm doing here is the same thing as component instances. All this change does is allow the CellInterface to directly look up the port attributes, without having to get the referenced primitive and check its attributes there.

Reply:
Ah, Yeah that makes sense. Thanks!